nemo/export/tensorrt_llm_parsing_utils.py

```diff
@@ -0,0 +1,204 @@
+        "-kv_fp8",
+        "--use_fp8_kv_cache",
+        default="auto",
+        type=str,
+        help="Enables exporting with FP8-quantizatized KV-cache",
+    )
+    return parser
+
+
+def parse_quantization_args(args):
+    args.export_fp8_quantized = str_to_bool("export_fp8_quantized", args.export_fp8_quantized, optional=True)
+    args.use_fp8_kv_cache = str_to_bool("use_fp8_kv_cache", args.use_fp8_kv_cache, optional=True)
+    return args
+
+
+def add_multi_block_mode_flag(parser):
+    parser.add_argument(
+        "-mbm",
+        '--multi_block_mode',
+        default=False,
+        action='store_true',
+        help='Split long kv sequence into multiple blocks (applied to generation MHA kernels). \
+                        It is beneifical when batchxnum_heads cannot fully utilize GPU. \
+                        Only available when using c++ runtime.',
+    )
+    return parser
+
+
+def add_export_flags(parser, deploy: bool):
+    parser.add_argument("-nc", "--nemo_checkpoint", required=(not deploy), type=str, help="Source .nemo file")
+    parser.add_argument(
+        "-mt",
+        "--model_type",
+        type=str,
+        required=(not deploy),
+        choices=["gptnext", "gpt", "llama", "falcon", "starcoder", "mixtral", "gemma"],
+        help="Type of the model. gptnext, gpt, llama, falcon, starcoder, mixtral and gemma are only supported."
+        " gptnext and gpt are the same and keeping it for backward compatibility",
+    )
+    parser.add_argument("-ng", "--num_gpus", default=None, type=int, help="Number of GPUs for the deployment")
+    parser.add_argument("-tps", "--tensor_parallelism_size", default=1, type=int, help="Tensor parallelism size")
+    parser.add_argument("-pps", "--pipeline_parallelism_size", default=1, type=int, help="Pipeline parallelism size")
+    parser.add_argument("-mil", "--max_input_len", default=256, type=int, help="Max input length of the model")
+    parser.add_argument("-mol", "--max_output_len", default=256, type=int, help="Max output length of the model")
+    parser.add_argument("-mbs", "--max_batch_size", default=8, type=int, help="Max batch size of the model")
+    parser.add_argument("-mnt", "--max_num_tokens", default=None, type=int, help="Max number of tokens")
+    parser.add_argument("-ont", "--opt_num_tokens", default=None, type=int, help="Optimum number of tokens")
+    parser.add_argument(
+        "-mpet", "--max_prompt_embedding_table_size", default=None, type=int, help="Max prompt embedding table size"
+    )
+    parser.add_argument(
+        "-npkc", "--no_paged_kv_cache", default=False, action='store_true', help="Enable paged kv cache."
+    )
+    parser.add_argument(
+        "-drip",
+        "--disable_remove_input_padding",
+        default=False,
+        action='store_true',
+        help="Disables the remove input padding option.",
+    )
+    parser.add_argument(
+        "-dt",
+        "--dtype",
+        choices=["bfloat16", "float16", "fp8", "int8"],
```

Comment:
Are "fp8" and "int8" really viable choices? These are quantization dtypes and would require some scaling factors to come along as well

Reply:
The `export_to_trt_llm.py` script, it throws an error right away:
https://github.com/NVIDIA/NeMo/blob/main/scripts/export/export_to_trt_llm.py#L161-L166

That's not a case with `deploy_triton.py` -- I just left the current behaviour unchanged

Reply:
I suggest we change this as it looks outdated and might be confusing CC @oyilmaz-nvidia 
1. Let's remove [this part](https://github.com/NVIDIA/NeMo/blob/110db0ca98dc1da1670421a73eee11c03039f22c/scripts/export/export_to_trt_llm.py#L161-L166) as it's no longer true: **"float16"** can also be used
2. I suggest we remove **"fp8"** and **"int8"** from `choices`. These are related to quantization, but anyway specifying either leads to immediate errors [here](https://github.com/NVIDIA/TensorRT-LLM/blob/a65dba7aaf7e2d8bb0120eea8f8f04deff145d6a/tensorrt_llm/plugin/plugin.py#L110)
3. This will be consistent with `dtype` mentioned [here](https://github.com/NVIDIA/NeMo/blob/110db0ca98dc1da1670421a73eee11c03039f22c/nemo/export/tensorrt_llm.py#L198) when used for a Nemo model